lib/utils/logDeprecation.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+const chalk = require('chalk');
+const memoizee = require('memoizee');
+
+const loggedDeprecations = {};
+
+function consoleLog(...args) {
+  console.log(...args); // eslint-disable-line no-console
+}
+
+function extractCodes(codesStr) {
+  if (!codesStr) {
+    return [];
+  }
+  return codesStr.split(',').reduce((acc, code) => {
+    acc[code] = true;
+    return acc;
+  }, {});
+}
+
+const prepareCodes = memoizee((serviceConfigCodesStr, envCodesStr) => {
+  const serviceConfigCodes = extractCodes(serviceConfigCodesStr);
+  const envCodes = extractCodes(envCodesStr);
+  const disabledCodes = Object.assign({}, envCodes, serviceConfigCodes);
+  return disabledCodes;
+});
+
+function writeDeprecation(message, docURL) {
+  consoleLog(`${chalk.red(message)}\n${chalk.dim(`More Info: ${docURL}`)}`);
+}
+
+function logDeprecation(code, message, docURL, serviceConfig) {
+  const disabledCodes = prepareCodes(
+    serviceConfig && serviceConfig.disabledCodes,
+    process.env.SLS_DEPRECATION_DISABLE
```

Comment:
This I would resolve upfront in modules body, and probably keep in `deprecatedByEnv` _set_ var

Reply:
I added it here because it was difficult to override process.env with specific value before certain test cases. Unless I re-require manually on these tests

Reply:
Yes, and you may override `process.env.SLS_DEPRECATION_DISABLE`  before re-requiring

Reply:
Done